### DISCUSSION TITLE
Filtering Data by Relationship Column

### DISCUSSION THREAD

**Discussion by stellarfinn** [Budibase/budibase#4645] [Help]
Hello, 

Budibase Cloud here.

I have two databases, Products and Brands. I'm displaying a catalog of products to our internal team. 

I'm setting this up so I can have a form to add new Brands to the Brands database as we bring on new vendors. 

I have an options picker on my catalog display which I've been able to link to the Brands database and populate with the existing Brands. 

However, I'm at a loss of how to get the Data Provider to filter results by the comparing the value provided by the picker to the primaryDisplay value of the Product 'vendor' relationship column. 

For a live view of what I'm doing please see https://ciaoimports.budibase.app/app/ciaob2b

Any help appreciated.

**Comment by stellarfinn** [comment]
@mjashanks Any thoughts here? I wasn't sure any previous discussions addressed this.

**Comment by mjashanks** [comment]
Hey @stellarfinn - apologies. I'm not clear on what you are asking for.

>  I'm at a loss of how to get the Data Provider to filter results

What table does the data provider read from

> ...by the comparing the value provided by the picker

What does the picker list? Brands?

Where do "vendors" come into it? is this another table?

**Reply by stellarfinn** [reply]
Hey @mjashanks - sorry for the confusion

I have to database tables - Brands (Vendors), and Products.

I wanted to be able to add new Brands in a simple form. I got that done. 
Then, I added a relationship O2M between Brands to Products.
This added a "Vendors" column in the Products database and provided a dropdown of Brands (Vendors) to select from when creating a new Product. 

When I went to display products on the front page, I could not get the Brands Options Picker dropdown to change the Products displayed depending on the Brand you chose. 

When filtering the Products data, there is no column to select the relationship column Vendors (which is populated by the Brands database) to compare the value selected by the Options Picker. 

Sorry if this is confusing. I attempted to figure it out on my own and it resulted in some unconventional database column naming as I tried.

**Reply by mjashanks** [reply]
@stellarfinn As i mentioned in another comment, filtering by relationship is something that we really want to implement, but have not done so yet.

However, I have managed to produce the scenario above. It may have performance issues though - but you can try!

So, my structure is as follows (app export below as well)

```
> Form
  > Data Provider : Brands (unfiltered)
     > Options picker : loading from the Brands data provider, Field = "selectedBrand"
  > Data Provider : Brands - filtered where _id equals selectedBrand, limit 1
     > Repeater
         > Data Provider : Relationships Repeater.Products.Vendors
             > Options Picker : loading data from the above data provider
```

So basically what happens here is
1. User selects a value from the Brands dropdown
2. The next data provider re-fetches that selected brand from the DB
3. The next data provider loads all products for that brand

[Products-export-1645807711853.txt](https://github.com/Budibase/budibase/files/8142885/Products-export-1645807711853.txt)

**Reply by stellarfinn** [reply]
@mjashanks Thanks for your reply. I was pretty sure this was the case. We'll be using this internally among a small team and down the road may open it up to B2B customers to have access to comprehensive product data. So I'll give your solution a try because the scale isn't really a problem at the moment. I appreciate your time and contributions! Overall, I think Budibase has GREAT potential to displace 95% of code for a large set of use cases. I've been able to build this in such a quick period it's really fantastic.

**Reply by mjashanks** [reply]
Thanks @stellarfinn  :) really appreciate it
